src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,112 @@
+package duke.storage;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.DukeTask;
+import duke.task.Event;
+import duke.task.ToDo;
+
+/**
+ * Encapsulates saving/loading data from storage.
+ *
+ * @author Kartikeya
+ */
+public class Storage {
+    private static final String FILE_FOLDER_NAME = "data";
+    private static final String FILENAME = "duke.txt";
+    private static final String FILEPATH = FILE_FOLDER_NAME + File.separator + FILENAME;
+
+    public Storage() throws DukeException {
+        this.createStorageFile();
+    }
+
+    private void createStorageFile() throws DukeException {
+        try {
+            File directory = new File(FILE_FOLDER_NAME);
+            if (!directory.exists()) {
+                directory.mkdir();
+            }
+            File saved = new File(FILEPATH);
+            if (!saved.exists()) {
+                saved.createNewFile();
+            }
```

Comment:
I like the way you avoid deep nesting a formatted your code neatly based on the different possible cases, shown above as well.